ui/src/app/settings/components/GeneralForm/GeneralForm.js

```diff
@@ -0,0 +1,96 @@
+import { Formik } from "formik";
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import * as Yup from "yup";
+
+import actions from "app/settings/actions";
+import selectors from "app/settings/selectors";
+import Button from "app/base/components/Button";
+import Form from "app/base/components/Form";
+import Loader from "app/base/components/Loader";
+import Row from "app/base/components/Row";
+import Col from "app/base/components/Col";
+import FormikField from "app/base/components/FormikField";
+
+const GeneralSchema = Yup.object().shape({
+  maas_name: Yup.string().required(),
+  enable_analytics: Yup.boolean()
+});
+
+const GeneralForm = () => {
+  const dispatch = useDispatch();
+  const updateConfig = actions.config.update;
+
+  const maasName = useSelector(selectors.config.maasName);
+  const analyticsEnabled = useSelector(selectors.config.analyticsEnabled);
+
+  return (
+    <Row>
+      <Col size="6">
+        <Formik
+          initialValues={{
+            maas_name: maasName,
+            enable_analytics: analyticsEnabled
+          }}
+          validationSchema={GeneralSchema}
+          onSubmit={(values, { setSubmitting }) => {
+            setTimeout(() => {
+              dispatch(updateConfig(values));
+              // Object.keys(values).forEach(key => {
+              //   dispatch(updateConfig(key, values[key]));
+              // });
+              setSubmitting(false);
+            }, 500);
```

Comment:
This timeout is fine for now, but just so you know in https://github.com/canonical-web-and-design/maas-ui/pull/63 I'll create an ActionButton component that has a success state. Also the action is taken immediately, only the UI is faked to make it seem like it's still saving.

Reply:
yeah, I was waiting on that before getting too involved with this button